<launch>
  <arg name="camera_name" default="axis_camera" />
  <arg name="hostname" default="192.168.0.228:8001" />
  <arg name="username" default="" />
  <arg name="password" default="" />
  <arg name="encrypt_password" default="false" />

  <arg name="enable_theora" default="false" />
  <arg name="enable_ptz" default="true" />
  <arg name="enable_ptz_teleop" default="true" />
  <arg name="enable_defog" default="false" />
  <arg name="enable_ir" default="false" />
  <arg name="enable_wiper" default="false" />
  <arg name="width" default="640" />
  <arg name="height" default="480" />
  <arg name="camera" default="0" />

  <group>
    <push_ros_namespace namespace="$(var camera_name)"/>
  </group>

    <node pkg="axis_camera" exec="axis" name="axis">
      <param name="hostname" value="$(var hostname)" />
      <param name="username" value="$(var username)" />
      <param name="password" value="$(var password)" />
      <param name="use_encrypted_password" value="$(var encrypt_password)" />
      <param name="width" value="$(var width)" />
      <param name="height" value="$(var height)" />
      <param name="camera" value="$(var camera)" />
      <param name="frame_id" value="$(var camera_name)" />
      <param name="wiper" value="$(var enable_wiper)" />
      <param name="defog" value="$(var enable_defog)" />
      <param name="ir" value="$(var enable_ir)" />
    </node>

    <node pkg="axis_camera" exec="axis_ptz" name="axis_ptz" if="$(var enable_ptz)">
      <param name="hostname" value="$(var hostname)" />
      <param name="username" value="$(var username)" />
      <param name="password" value="$(var password)" />
      <param name="use_encrypted_password" value="$(var encrypt_password)" />
      <param name="width" value="$(var width)" />
      <param name="height" value="$(var height)" />
      <param name="camera" value="$(var camera)" />
      <param name="frame_id" value="$(var camera_name)" />
      <param name="wiper" value="$(var enable_wiper)" />
      <param name="defog" value="$(var enable_defog)" />
      <param name="ir" value="$(var enable_ir)" />
    </node>

<!--
    <node pkg="image_transport" exec="republish" name="republish"
          args="compressed" if="$(var enable_theora)">
      <remap from="in" to="image_raw" />
      <remap from="out" to="image_raw_out" />
    </node>
-->

    <node pkg="axis_camera" exec="axis_teleop" name="axis_teleop" if="$(var enable_ptz_teleop)">
      <param name="use_enable_button" value="false" />
      <param name="enable_button" value="4" />
      <param name="scale_pan_deg" value="25" />
      <param name="scale_tilt_deg" value="25" />
</node>

  <!-- To control the PTZ, issue commands like so:
  rostopic pub /axis/cmd axis_camera/Axis '{ brightness: 5000, pan: 100, tilt: 10, zoom: 0, autofocus: True }'
  NOTE: Don't forget to populate the brightness field, or
        you'll get very dark images.
  -->
</launch>
